presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -2718,6 +2721,29 @@ public void testMaterializedPartitioning()
             assertUpdate("DROP TABLE IF EXISTS test_bucketed_lineitem1");
             assertUpdate("DROP TABLE IF EXISTS test_bucketed_lineitem2");
         }
+
+        // Window functions
+        assertQuery(
+                materializeExchangesSession,
+                "SELECT sum(rn) FROM (SELECT row_number() OVER(PARTITION BY orderkey) as rn FROM lineitem)",
+                "SELECT 180782",
+                assertRemoteMaterializedExchangesCount(1)
+                        // make sure that the window function has been planned as a RowNumberNode
+                        .andThen(plan -> assertTrue(searchFrom(plan.getRoot()).where(node -> node instanceof RowNumberNode).matches())));
+        assertQuery(
+                materializeExchangesSession,
+                "SELECT sum(rn) FROM (SELECT row_number() OVER(PARTITION BY orderkey ORDER BY linenumber) as rn FROM lineitem) WHERE rn < 5",
+                "SELECT 107455",
+                assertRemoteMaterializedExchangesCount(1)
+                        // make sure that the window function has been planned as a TopNRowNumberNode
+                        .andThen(plan -> assertTrue(searchFrom(plan.getRoot()).where(node -> node instanceof TopNRowNumberNode).matches())));
+        assertQuery(
```

Comment:
Let's put these three tests in the same sequence as the visitor class: `Window` -> `RowNumber` -> `TopNRowNumber`.